upt_macports/upt_macports.py

```diff
@@ -30,9 +30,25 @@ def licenses(self):
         filepath = pkg_resources.resource_filename(__name__, relpath)
         with open(filepath) as f:
             spdx2macports = json.loads(f.read())
+        licenses = []
+        if not self.upt_pkg.licenses:
+            self.logger.info('No license found')
+            return 'unknown'
 
-        return ' '.join([spdx2macports.get(license.spdx_identifier, 'unknown')
-                        for license in self.upt_pkg.licenses]) or 'unknown'
+        for license in self.upt_pkg.licenses:
+            try:
+                if license.spdx_identifier == 'unknown':
+                    port_license = 'unknown'
+                    self.logger.warning(f'upt failed to detect license')
+                else:
+                    port_license = spdx2macports[license.spdx_identifier]
+                    self.logger.info(f'Found license {port_license}')
+                licenses.append(port_license)
+            except KeyError:
+                err = f'Unknown license {license.spdx_identifier}'
+                self.logger.error(err)
+                self.logger.info('Please report the error at https://github.com/macports/upt-macports') # noqa
```

Comment:
if we end up in this ```except``` block, it means that ```upt``` found a license but it's not converted yet into a license that MacPorts understands (i.e., is present in ```spdx2macports.json```). 

So if I read this correctly that keeps ```licenses = []```, as initialized and logging it as an error will not halt the script, right? So now if we do the return statement it will return ```''``` and put the line ```license ''``` in the Portfile? I don't think that's what we want (or at least what I prefer). What about making the error message a bit more specific and return that same message so that it will be added in the ```license``` field in the Portfile as well? Perhaps something like:
```err = f'MacPorts license unknown for {license.spdx_identifier}'```